mxcube3/ui/components/SampleView/ContextMenu.js

```diff
@@ -105,10 +105,10 @@ export default class ContextMenu extends React.Component {
     const { sampleID, defaultParameters, shape, sampleData } = this.props;
     const sid = _shape ? _shape.id : shape.id;
 
-    if (this.props.clickCentring) {
-      this.props.sampleActions.stopClickCentring();
```

Comment:
This seems strange to me, why do we need to stop a click centering which is not started ?

The 2D centring routine should either start a click centring, or if its not really a click centering but translation of sampx sampy, just do sendAddShape or maybe sendAcceptCentring.

I dont know exactly how your 2D centring is implemented, is it simply a call to move to beam ?

Reply:
Our 2d is a move_to_beam. 

This allows to add a task and save the point in the middle of the centring procedure, on a temporary point.

Reply:
no no, we do a 2d centring, get a temp point, and on  that try to add a task. So it should also save the point automatically. No matter if you are in a centring procedure

Reply:
So then its maybe enough to just save the position, id like that we avoid calling acceptCentring if its not necessary, especially since the route handling acceptCentring performs no checking before the call is made.

So I see two options, guard the acceptCentring route or just make a call to updateShape